src/lib/optimizer/optimizer.cpp

```diff
@@ -97,6 +98,8 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
 
   optimizer->add_rule(std::make_unique<PredicateMergeRule>());
 
+  optimizer->add_rule(std::make_unique<NullScanRemovalRule>());
```

Comment:
Any reason to add it this late? Shouldn't we execute it as early as possible so that the removed predicates do not participate in predicate ordering?

Reply:
I did not think about it. I moved the rule to the beginning.

Reply:
I'd move it below `PredicateSplitUpRule()` (not `PredicateSplitUpRule(false)`) so that you can also catch predicates that are only "free" after a split-up.